examples/platforms/cc2538/radio.c

```diff
@@ -277,9 +349,60 @@ void cc2538RadioInit(void)
     HWREG(RFCORE_XREG_TXPOWER) = sTxPowerTable[0].mTxPowerReg;
     sTxPower                   = sTxPowerTable[0].mTxPowerVal;
 
+#if OPENTHREAD_CONFIG_CC2538_WITH_CC2592
+    // PA_EN pin configuration.
+    // Step 1. make it an output
+    HWREG(GPIO_C_BASE | GPIO_O_DIR) |= GPIO_PIN(OPENTHREAD_CONFIG_CC2592_PA_EN_PIN);
+    // Step 2. Route PA_PD to OBS0 and invert it to produce PA_EN
+    HWREG_ARR(RFCORE_XREG_RFC_OBS_CTRL, 0) = RFCORE_XREG_RFC_OBS_POL_INV      // Invert the output
+                                             | RFCORE_XREG_RFC_OBS_MUX_PA_PD; // PA "power down" signal
+    // Step 3. Connect the selected pin to OBS0 and enable OBS0.
+    HWREG_ARR(CCTEST_OBSSEL, OPENTHREAD_CONFIG_CC2592_PA_EN_PIN) = CCTEST_OBSSEL_EN          // Enable the output
+                                                                   | CCTEST_OBSSEL_SEL_OBS0; // Select OBS0
+
+    // LNA_EN pin configuration.
+    HWREG(GPIO_C_BASE | GPIO_O_DIR) |= GPIO_PIN(OPENTHREAD_CONFIG_CC2592_LNA_EN_PIN);
+    HWREG_ARR(RFCORE_XREG_RFC_OBS_CTRL, 1) = RFCORE_XREG_RFC_OBS_POL_INV | RFCORE_XREG_RFC_OBS_MUX_LNA_PD;
+    HWREG_ARR(CCTEST_OBSSEL, OPENTHREAD_CONFIG_CC2592_LNA_EN_PIN) = CCTEST_OBSSEL_EN | CCTEST_OBSSEL_SEL_OBS1;
+
+#if OPENTHREAD_CONFIG_CC2592_USE_HGM
+    // HGM pin configuration.  Set the pin state first so we don't glitch.
+    cc2538RadioSetHgm(OPENTHREAD_CONFIG_CC2592_HGM_DEFAULT_STATE);
+    HWREG(OPENTHREAD_CONFIG_CC2592_HGM_PORT | GPIO_O_DIR) |= GPIO_PIN(OPENTHREAD_CONFIG_CC2592_HGM_PIN);
+#endif // OPENTHREAD_CONFIG_CC2592_USE_HGM
+#endif // OPENTHREAD_CONFIG_CC2538_WITH_CC2592
+
     otLogInfoPlat("Initialized", NULL);
 }
 
+#if OPENTHREAD_CONFIG_CC2538_WITH_CC2592 && OPENTHREAD_CONFIG_CC2592_USE_HGM
+void cc2538RadioSetHgm(bool aState)
+{
+    if (aState)
+    {
+        HWREG_ARR(OPENTHREAD_CONFIG_CC2592_HGM_PORT, GPIO_PIN(OPENTHREAD_CONFIG_CC2592_HGM_PIN)) =
+            GPIO_PIN(OPENTHREAD_CONFIG_CC2592_HGM_PIN);
+    }
+    else
+    {
+        HWREG_ARR(OPENTHREAD_CONFIG_CC2592_HGM_PORT, GPIO_PIN(OPENTHREAD_CONFIG_CC2592_HGM_PIN)) = 0;
+    }
+}
+
+bool cc2538RadioGetHgm()
```

Comment:
nit: All void functions or methods shall explicitly declare and specify the void type keyword.

Reply:
Ahh, too used to languages where `foo()` means a function that takes no arguments.  (or ones that don't care like Perl and JavaScript where they're accessed through `@_` or `arguments` respectively.)

Hopefully fixed in 2662163.